modules/file/doc.rst

```diff
@@ -163,8 +163,52 @@ Several options can be applied to a search:
 
 * **matchall** indicates that within a given search, all search filters must
   match on one file for it to be included in the results. Being a boolean,
-  `matchall` is not set by default. The MIG command line sets it automatically,
-  the console does not.
+  `matchall` is not set by default, but the command line and the console set it
+  when creating file searches. Use `matchany` to deactivate it. `matchall` has
+  a strong impact on search performances. See "Search algorithm".
+
+  Examples:
+	* `-name vim -sha1 21345asd -matchall` -> (name=vim AND sha1=21345asd)
+	* `-name vim -sha1 21345asd -matchany` -> (name=vim OR sha1=21345asd)
+
+* **macroal** stands for "Match All Contents Regexes On All Lines". It's a boolean
+  option that requires that all `content` regexes must match on all the lines of
+  a file. By default, content regexes are applied at the file level and will
+  return a match if one line matches one regex, and if another line matches another
+  regex. When the `macroal` option is set, each line in the file must match all
+  content regexes defined in a given search to return a match. It is set to not
+  set by default.
+
+  example: `-path /home -name authorized_keys -content "^(#|\s+|...list of ssh keys...)$" -macroal`
+
+  will list authorized_keys file that have contain either a comment, an empty
+  line or one of the listed ssh keys. If a file has a line that doesn't match
```

Comment:
nit: clarifications also as this option seem rather special:

If the file has ANY line that doesn't match

It may be useful to insert that this is here as a different way to do a "not" match/search (which Im guessing is the intent), so instead of "not matching x" or doing a "grep -v" you can do a very explicit selection in the file